app/src/main/java/com/minimalisticapps/priceconverter/data/remote/dto/BitPayExchangeRate.kt

```diff
@@ -4,24 +4,28 @@ import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
 import com.google.gson.annotations.SerializedName
+import java.math.BigDecimal
 
 @Entity(tableName = "Coins")
 data class BitPayExchangeRate(
     @ColumnInfo(name = "code")
     var code: String,
+
     @ColumnInfo(name = "rate")
     @SerializedName("value")
-    var rate: Double?,
+    var rate: BigDecimal?,
+
     @PrimaryKey(autoGenerate = false)
     @ColumnInfo(name = "name")
     val name: String,
+
     @ColumnInfo(name = "type")
     val type: String,
+
     @ColumnInfo(name = "unit")
     @SerializedName("unit")
     val unit: String,
+
     @ColumnInfo(name = "one_shit_coin_value")
-    var oneShitCoinValue: Double?,
```

Comment:
Moved to separate state. It is independent of the exchange rates from the BitPayExchange.

The model here needs to be refactored even more in the future.

This was first step. I have separated the value of the input of the given currency from the exchange rates and Shitcoins' metadata